News-Api/src/Articles.jsx

```diff
@@ -1,48 +1,50 @@
 import { getArticles } from "./api.js";
-import { useState, useEffect, useContext} from "react";
-import { Link } from "react-router-dom";
-import ArticleListCard from './ArticleListCard.jsx';
+import { useState, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import ArticleListCard from "./ArticleListCard.jsx";
 import Header from "./Header.jsx";
 import { dateFormatter } from "./dateUtils.js";
 
-
 const Home = () => {
-
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-
+  const [isError, setIsError] = useState(false);
+  const { topic } = useParams();
 
   useEffect(() => {
-    getArticles().then((articleData) => {
-     
-          setArticles(articleData);
-
-      setIsLoading(false);
-    });
-  }, []);
+    getArticles(topic)
+      .then((articleData) => {
+        setIsLoading(true);
+        setArticles(articleData);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setIsError(true);
+        console.log(error, "error");
+      });
+  }, [topic]);
 
   if (isLoading) return <p>Loading...</p>;
+  if (isError) return <p>Something went wrong...</p>;
 
   return (
     <section className="Articles">
-      <Header title='Article List'/>
-          <ul className="Article_list">            
-        {articles.map(({ article_id, title, author, topic, comment_count, created_at }) => (
-          
-           <Link key={article_id} to={`/articles/${article_id}`}>
-            <ArticleListCard key={article_id} 
-            title={title} 
-            author={author}
-            topic={topic}
-            comment_count={comment_count}
-       
-            created_at={dateFormatter(created_at) 
-
-          }
-            
-           />
-          </Link>
-        ))}
+      <Header title="Article List" />
```

Comment:
Could maybe name this articles?
Or even use the params to explain that its coding articles for example? up to you.